### ISSUE TITLE
HRC field models are much larger than what QGears expects

### ISSUE THREAD

**Issue by paulsapps** [q-gears/q-gears#115]
The converted HRC models are far too large. Use the data installer from my "installer" branch and replace all field models. You can make out a giant head bigger than the screen.

Somehow the converted BSX PSX models are the correct size, need to figure out what must be done to cure this.

**Comment by paulsapps** [comment]
@myst6re @zerotacg guys, any ideas? The animations appear to be incorrect too. When I scaled down by a factor of 12 when cloud walks his fee are "in" the camera. Also as you can see from the screen shot the colours are not quite correct.

**Comment by myst6re** [comment]
Yes I divide file values (bone lengths, animation translations and vertex points) by:
 - 4096.0 in PS version
 - 132.0 in PC version

It seems it is divided by 1024 on q-gears (PS dump)
So you need to divide by ~33 for the PC version.

There is also another difference on PC: when I build the model, I do the rotation, I draw a part, and I translate to another bone. On PS version the translation is before the rotation.

Compare bones values, you will see the shift.

**Comment by zerotacg** [comment]
Not sure about the size, last time I used it the PC model loading was just fine.
Colours could be due to lighting. I implemented serialization of normals for PC models which results in shading, as opposed to the flat colours of the exported psx models.

**Comment by paulsapps** [comment]
@myst6re not sure what I need to change for the "draw a part and then translate" since we just create an ogre mesh file? @zerotacg yes in the test app its fine, in the actual engine it goes crazy as in my screen shots :)

I will try to scale by 33 later, but I guess translate/rotate issue will still be there I can try disabling normals too.

**Comment by Akari1982** [comment]
> as opposed to the flat colours of the exported psx models.

PSX version doesnt use flat colours. It calcilates light for each model during field loading. And it can be changed dynamicly by using KAWAI opcode.

**Comment by paulsapps** [comment]
So normals are not the reason PC models are looking much brighter than PSX?

**Comment by paulsapps** [comment]
@zerotacg @myst6re been a while, any other ideas? It seems that the whole model needs rotating on one axis by 90 degrees. But anytime I try this in the conversion the model ends up really messed up. I can't continue script conversion until this is fixed because it makes testing impossible :(

**Comment by zerotacg** [comment]
I don't know what you men by "test app" but I meant in the real engine like in the scene you posted
I was loading the models from original pc data with adjusted map xml like so

````
<entity_model name="Cloud" file_name="aaaa.mesh" position="29.0547 229.234 2.56082"
    direction="0"
    scale="0.03125 0.03125 0.03125"
    root_orientation="0.7071067811865476 0.7071067811865476 0 0" />
````

that should give you the right scale and the quaternion for correct rotation

**Comment by paulsapps** [comment]
where did root_orientation come from? by test app I mean the tool that converts them

**Comment by paulsapps** [comment]
This root_orientation fixes the rotation but the translation of some animations seems to be wrong? For example when they jump off the train they fly in from the left side of the screen instead of the right :s

**Comment by paulsapps** [comment]
Update bug title to reflect the actual remaining issue.

**Comment by paulsapps** [comment]
Seems that set_rotation was always setting 0 hence the incorrect orientation. However still one more problem as always. Now the translation of animations appears to be wrong. This time it isn't a script issue as I've compared the original scripts VS the SUDM generated scripts and the set position calls have equal parameters.

So for example when a guard is kicked he flys away from rather than towards the camera, then jumps to the correct location when the "dead" animation is set. Needs more investigation as to what is causing this and the wrong colours. But this is now working enough for me to progress on with more script fixes.
